game/end_to_end_tests/game_page.py

```diff
@@ -86,7 +86,10 @@ def step(self):
 
     def solution_button(self):
         self.browser.find_element_by_id("solution_tab").click()
-        time.sleep(1)
+        popup_message = self.browser.find_element_by_id("myModal-title").text
+        while not popup_message == "View solution":
```

Comment:
not sure if popups are good there as teachers will see it outside of tests too .... we could add a parameter but then the tests won't test the "actual" feature...